mesonbuild/interpreter.py

```diff
@@ -2807,6 +2840,15 @@ def _find_cached_fallback_dep(self, name, dirname, varname, wanted, required):
         dep = self.get_subproject_dep(name, dirname, varname, required)
         if not dep:
             return False
+        if is_disabler(dep):
+            if required:
+                if not name:
+                    name = '%s/%s' % (dirname, varname)
+                raise DependencyException('%s disabled by %s subproject' % (name, dirname))
+            mlog.log('Dependency', mlog.bold(name), 'from subproject',
+                     mlog.bold(dirname), 'found:', mlog.red('NO'), '(subproject disabled)')
+            return dep
```

Comment:
I think this whole block is not needed if you do the check in get_subproject_dep() already. Otherwise it would be needed in dependency_fallback() too.

Reply:
I implemented it the way it is because of how `    def dependency_fallback(self, name, kwargs):` is implemented. It does the right thing I believe.

Reply:
I think `get_subproject_dep()` can return 3 things:
- A **found** dep object if the subproject works
- `None` if the subproject does not work and required=false
- Raise exception of subproject does not work and required=true

To achieve that, I think the proper way is for get_subproject_dep() to raise exception when subproject.found() is false, which will make it enter the `except:` block. Because a not-found subproject should be the same thing as an undefined subproject, I think.

If you do that, then _find_cached_fallback_dep() does not need any changes. Also check dependency_fallback(), with your code it would return a not-found subproject even if required=true.